scripts/migrateAttendance.js

```diff
@@ -0,0 +1,178 @@
+// This file is to help migrate the old database which had users with an 'attendance' field
+// to a new attendance collection
+
+// var users = db.users.find({});
+//
+// while(users.hasNext()) {
+//     for
+//     db.attendances.create({})
+//     db.projects.update(
+//         { "repositoryUrl": { $exists: true } },
+//         {
+//             $set: { repositories: [ projectsWithRepoUrl.next().repositoryUrl ] },
+//             $unset: { repositoryUrl: 1 }
+//         }
+//     );
+// }
+
+function standardDate(d){
+  var date = new Date(d);
+  date.setHours(0,0,0,0);
+  return date;
+}
+
+var smallgroups = db.smallgroups.find({});
+
+while(smallgroups.hasNext()) {
+    smallgroups.forEach(function(group){
+        if (group.semester){
+            var classyear = db.classyears.findOne({semester:group.semester})
+            db.smallgroups.update(
+                { "_id": group._id },
+                {
+                    $set: { "classYear": classyear._id },
+                    $unset: { semester: 1 }
+                }
+            );
+        }
+    })
+
+}
+
+var users = db.users.find({});
+var classyear = db.classyears.findOne({current:true});
+
+while(users.hasNext()){
+    users.forEach(function(user){
+        var attendance = user.attendance;
+        var unverifiedAttendance = user.unverifiedAttendance;
+        var smallgroup = false;
+
+        if (user.smallgroup){
+            smallgroup = db.smallgroups.findOne({"_id":user.smallgroup});
+        }
+        var allfound = true;
+        if (attendance){
+            for (var a = 0; a < attendance.length ; a++){
+                var date = attendance[a];
+                var found  = false;
+                for (var b = 0; b < classyear.dayCodes.length ; b++){
+                    if (standardDate(classyear.dayCodes[b].date).getTime() == standardDate(date).getTime()){
+                        var daycode = classyear.dayCodes[b];
+                        db.attendances.insert(
+                            {
+                                classYear: classyear._id,
+                                user: user._id,
+
+                                date: standardDate(date),
```

Comment:
We still keep this in datetime. Date is the date of the attendance, and for ease of db queries